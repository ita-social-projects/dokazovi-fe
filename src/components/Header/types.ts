export interface IHeaderProps {
  id: string;
  label: string;
  url: string;
}
