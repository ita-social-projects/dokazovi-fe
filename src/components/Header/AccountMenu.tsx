import React, { useContext, useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { Avatar, Button, Typography } from '@material-ui/core';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import { useTranslation } from 'react-i18next';
import { useStyles } from './AccountMenu.styles';
import {
  StyledMenu,
  StyledMenuItem,
} from '../../old/lib/components/Menu/StyledMenu';
import {
  getUserAsyncAction,
  selectCurrentUser,
  signOutAction,
} from '../../models/user';
import { useActions } from '../../shared/hooks';
import { AuthContext } from '../../old/provider/AuthProvider/AuthContext';
import { AccountIcon } from '../../old/lib/components/icons/AccountIcon';
import { langTokens } from '../../locales/localizationInit';
import { clearAuthoritiesAction } from '../../models/authorities/reducers';
import { useCheckAdmin } from '../../old/lib/hooks/useCheckAdmin';

export const AccountMenu: React.FC = () => {
  const { t } = useTranslation();

  const classes = useStyles();
  const user = useSelector(selectCurrentUser);
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const { authenticated } = useContext(AuthContext);
  const { removeAuthorization } = useContext(AuthContext);
  const [boundSignOutAction] = useActions([signOutAction]);
  const [boundGetUserAsyncAction] = useActions([getUserAsyncAction]);
  const [boundClearAuthorities] = useActions([clearAuthoritiesAction]);
  const authorities = useCheckAdmin();

  const onLogoutHandler = () => {
    boundSignOutAction();
    removeAuthorization();
    boundClearAuthorities();
  };

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  useEffect(() => {
    if (authenticated) {
      boundGetUserAsyncAction();
    }
  }, [authenticated]);

  if (!user.data) {
    return null;
  }

  return (
    <>
      <Button
        color="primary"
        aria-controls="simple-menu"
        aria-haspopup="true"
        onClick={handleClick}
        className={classes.button}
      >
        {user.data ? (
          <Avatar
            alt={`${user.data.firstName} ${user.data.lastName}`}
            className={classes.avatar}
            src={user.data.avatar}
          />
        ) : (
          <AccountIcon className={classes.avatar} />
        )}
        {user.data && <ArrowDropDownIcon className={classes.icon} />}
      </Button>
      <StyledMenu
        id="account-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
        className={classes.menu}
      >
        <Link to="/profile">
          <StyledMenuItem onClick={handleClose}>
            <Typography variant="button" color="inherit">
              {t(langTokens.common.profile)}
            </Typography>
          </StyledMenuItem>
        </Link>
        {authorities ? (
          <Link to="/admin">
            <StyledMenuItem onClick={handleClose}>
              <Typography variant="button" color="inherit">
                {t(langTokens.common.admin)}
              </Typography>
            </StyledMenuItem>
          </Link>
        ) : null}
        <Link to="/">
          <StyledMenuItem onClick={onLogoutHandler}>
            <Typography variant="button" color="inherit">
              {t(langTokens.common.exit)}
            </Typography>
          </StyledMenuItem>
        </Link>
      </StyledMenu>
    </>
  );
};
