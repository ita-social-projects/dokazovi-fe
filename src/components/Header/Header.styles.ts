import makeStyles from '@material-ui/core/styles/makeStyles';

export const useStyles = makeStyles(
  (theme) => ({
    header: {
      height: 120,
      backgroundColor: theme.palette.common.black,
    },
    headerMobile: {
      height: 85,
      backgroundColor: theme.palette.common.black,
    },
    container: {
      height: '100%',
      padding: theme.spacing(0, 10),
      [theme.breakpoints.down('xs')]: {
        padding: theme.spacing(0, 5),
      },
    },
    toolbar: {
      height: '100%',
      justifyContent: 'space-between',
      padding: 0,
    },
    logo: {
      color: theme.palette.common.white,
      marginBottom: 6,
    },
    actionsContainer: {
      display: 'flex',
      columnGap: theme.spacing(4),
    },
    tabs: {
      height: 55,
      marginLeft: '6vw',
      display: 'flex',
      alignItems: 'center',
      backgroundColor: theme.palette.common.black,
      [theme.breakpoints.down('md')]: {
        margin: '0 3vw 0 4vw',
      },
    },
    tab: {
      color: theme.palette.primary.main,
      fontWeight: 600,
      '&:not(:last-child)': {
        marginRight: theme.spacing(5),
      },
      '&.active': {
        color: theme.palette.common.white,
        fontWeight: 700,
      },
    },
    tabLabel: {
      fontSize: 18,
      color: 'inherit',
      fontWeight: 'inherit',
    },
    logoMobile: {
      fontSize: 38,
      color: theme.palette.common.white,
    },
    searchIcon: {
      color: theme.palette.common.white,
      fontSize: 48,
    },
    searchInputWrapper: {
      padding: 0,
    },
    searchInput: {
      width: '85%',
      height: 39,
      fontSize: 16,
      paddingLeft: theme.spacing(2),
    },
    searchInputIcon: {
      backgroundColor: theme.palette.common.white,
      marginRight: theme.spacing(2),
      fontSize: 32,
    },
    paper: {
      width: '70%',
    },
    label: {
      padding: 0,
    },
  }),
  { name: 'Header' },
);
