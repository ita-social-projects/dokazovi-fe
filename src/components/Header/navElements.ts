import { IHeaderProps } from './types';
import i18n, { langTokens } from '../../locales/localizationInit';

export const navElements: IHeaderProps[] = [
  {
    id: 'main',
    label: i18n.t(langTokens.common.main),
    url: '/',
  },
  {
    id: 'materials',
    label: i18n.t(langTokens.common.materials),
    url: '/materials',
  },
  {
    id: 'experts',
    label: i18n.t(langTokens.common.experts),
    url: '/experts',
  },
];
