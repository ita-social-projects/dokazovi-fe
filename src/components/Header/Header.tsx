import React, { useContext, useState } from 'react';
import {
  Box,
  Button,
  Container,
  Menu,
  MenuItem,
  Toolbar,
  Typography,
} from '@material-ui/core';
import { Link, NavLink, Route } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import SearchIcon from '@material-ui/icons/Search';
import { useStyles } from './Header.styles';
import { PostCreationMenu } from './PostCreationMenu';
import { LoginModal } from '../../old/lib/components/Users/LoginModal';
import { AccountMenu } from './AccountMenu';
import { AuthContext } from '../../old/provider/AuthProvider/AuthContext';
import i18n, { langTokens } from '../../locales/localizationInit';
import { IHeaderProps } from './types';
import { ScreenContext } from '../../old/provider/MobileProvider/ScreenContext';
import { BurgerMenu } from '../BurgerMenu/BurgerMenu';

export const navElems: IHeaderProps[] = [
  {
    id: 'main',
    label: i18n.t(langTokens.common.main),
    url: '/',
  },
  {
    id: 'materials',
    label: i18n.t(langTokens.common.materials),
    url: '/materials',
  },
  {
    id: 'experts',
    label: i18n.t(langTokens.common.experts),
    url: '/experts',
  },
];

export const Header: React.FC = () => {
  const classes = useStyles();
  const { t } = useTranslation();
  const { authenticated } = useContext(AuthContext);

  const [searchInput, setSearchInput] = useState<null | Element>(null);
  const [mobileMenuOpen, setMobileMenuOpen] = useState<boolean>(false);

  const { mobile } = useContext(ScreenContext);

  return (
    <div id="header" className={classes.header}>
      <Container className={classes.container}>
        <Toolbar className={classes.toolbar}>
          {mobile && <BurgerMenu
            navigation={navElems}
            mobileMenuOpen={mobileMenuOpen}
            setMobileMenuOpen={(b)=>setMobileMenuOpen(b)}
          />}
          <Box display="flex">
            {!mobileMenuOpen && <Link to='/'>
              <Typography className={mobile ? classes.logoMobile : classes.logo} variant='h1'>
                {t(langTokens.common.projectName)}
              </Typography>
            </Link>}
            {!mobile && <Box className={classes.tabs}>
              {navElems.map((item) => (
                <NavLink
                  to={item.url}
                  key={item.id}
                  className={classes.tab}
                  exact
                >
                  <Typography variant='h5' className={classes.tabLabel}>
                    {item.label}
                  </Typography>
                </NavLink>
              ))}
            </Box>}
          </Box>
          {!mobile && <Box className={classes.actionsContainer}>
            {authenticated && <PostCreationMenu />}
            {authenticated ? (
              <AccountMenu />
            ) : (
              <Route path='/opendoctorgate'>
                <LoginModal />
              </Route>
            )}
          </Box>}
          {mobile && <div>
            <Button aria-controls="simple-menu" aria-haspopup="true"
                    onClick={(e)=>setSearchInput(e.currentTarget)}>
              <SearchIcon className={classes.searchIcon} />
            </Button>
            <Menu
              id="simple-menu"
              anchorEl={searchInput}
              keepMounted
              open={Boolean(searchInput)}
              onClose={()=>setSearchInput(null)}
            >
              <MenuItem>
                <input className={classes.searchInput}/>
                <SearchIcon/>
              </MenuItem>
            </Menu>
          </div>}
        </Toolbar>
      </Container>
    </div>
  );
};
