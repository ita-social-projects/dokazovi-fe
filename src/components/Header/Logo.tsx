import React from 'react';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { langTokens } from '../../locales/localizationInit';
import { useStyles } from './Header.styles';

export const Logo = ({ mobile, isSearchVisible }) => {
  const { t } = useTranslation();
  const classes = useStyles();
  if (isSearchVisible) return null;

  return (
    <>
      <Box display="flex">
        <Link to="/">
          <Typography
            className={mobile ? classes.logoMobile : classes.logo}
            variant="h1"
          >
            {t(langTokens.common.projectName)}
          </Typography>
        </Link>
      </Box>
    </>
  );
};
