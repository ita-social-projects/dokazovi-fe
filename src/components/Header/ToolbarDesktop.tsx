import React, { useContext } from 'react';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import { NavLink, Route } from 'react-router-dom';
import { useStyles } from './Header.styles';
import { AuthContext } from '../../old/provider/AuthProvider/AuthContext';
import { PostCreationMenu } from './PostCreationMenu';
import { AccountMenu } from './AccountMenu';
import { LoginModal } from '../../old/lib/components/Users/LoginModal';
import { Logo } from './Logo';
// eslint-disable-next-line import/no-cycle
import { navElems } from './Header';

export const ToolbarDesktop = () => {
  const classes = useStyles();
  const { authenticated } = useContext(AuthContext);
  const mobile = false;
  const isSearchVisible = false;

  return (
    <div className={classes.toolbarDesktop}>
      <div style={{ display: 'flex' }}>
        <Logo mobile={mobile} isSearchVisible={isSearchVisible} />
        <Box className={classes.tabs}>
          {navElems.map((item) => (
            <NavLink to={item.url} key={item.id} className={classes.tab} exact>
              <Typography variant="h5" className={classes.tabLabel}>
                {item.label}
              </Typography>
            </NavLink>
          ))}
        </Box>
      </div>
      <Box className={classes.actionsContainer}>
        {authenticated && <PostCreationMenu />}
        {authenticated ? (
          <AccountMenu />
        ) : (
          <Route path="/opendoctorgate">
            <LoginModal />
          </Route>
        )}
      </Box>
    </div>
  );
};
