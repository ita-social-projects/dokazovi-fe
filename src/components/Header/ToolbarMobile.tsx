import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import SearchIcon from '@material-ui/icons/Search';
import { useStyles } from './Header.styles';
import { Logo } from './Logo';
import { BurgerMenu } from '../BurgerMenu/BurgerMenu';
import { navElements } from './navElements';

export const ToolbarMobile: React.FC = () => {
  const classes = useStyles();
  const [isSearchVisible, setIsSearchVisible] = useState<boolean>(false);
  const [mobileMenuOpen, setMobileMenuOpen] = useState<boolean>(false);

  return (
    <>
      <BurgerMenu
        navigation={navElements}
        mobileMenuOpen={mobileMenuOpen}
        setMobileMenuOpen={(b) => setMobileMenuOpen(b)}
      />
      <Logo isOnMobile isSearchVisible={isSearchVisible} />
      {isSearchVisible ? (
        <>
          <ClickAwayListener onClickAway={() => setIsSearchVisible(false)}>
            <TextField
              fullWidth
              variant="standard"
              className={classes.searchInput}
              InputProps={{
                endAdornment: (
                  <InputAdornment position="end">
                    <SearchIcon className={classes.searchInputIcon} />
                  </InputAdornment>
                ),
                disableUnderline: true,
              }}
            />
          </ClickAwayListener>
        </>
      ) : (
        <>
          <Button
            aria-controls="simple-menu"
            aria-haspopup="true"
            onClick={() => setIsSearchVisible((prev) => !prev)}
          >
            <SearchIcon className={classes.searchIcon} />
          </Button>
        </>
      )}
    </>
  );
};
