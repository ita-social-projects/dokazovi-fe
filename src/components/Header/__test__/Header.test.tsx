import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { store } from '../../../models/store';

import { Header } from '../Header';

test('header renderers', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
