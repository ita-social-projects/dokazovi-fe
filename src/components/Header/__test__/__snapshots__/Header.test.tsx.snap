// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests header should be rendered with "create" and "profile" if authenticated 1`] = `
<DocumentFragment>
  <div
    class="Header-header-20"
    id="header"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiContainer-root Header-container-22 MuiContainer-maxWidthLg"
    >
      <div
        class="Header-toolbarDesktop-24"
      >
        <div
          style="display: flex;"
        >
          <div
            class="MuiBox-root MuiBox-root-36"
          >
            <a
              href="/"
            >
              <h1
                class="MuiTypography-root Header-logo-25 MuiTypography-h1"
              >
                Доказові
              </h1>
            </a>
          </div>
          <div
            class="MuiBox-root MuiBox-root-37 Header-tabs-27"
          >
            <a
              aria-current="page"
              class="Header-tab-28 active"
              href="/"
            >
              <h5
                class="MuiTypography-root Header-tabLabel-29 MuiTypography-h5"
              >
                Головна
              </h5>
            </a>
            <a
              class="Header-tab-28"
              href="/materials"
            >
              <h5
                class="MuiTypography-root Header-tabLabel-29 MuiTypography-h5"
              >
                Матеріали
              </h5>
            </a>
            <a
              class="Header-tab-28"
              href="/experts"
            >
              <h5
                class="MuiTypography-root Header-tabLabel-29 MuiTypography-h5"
              >
                Автори
              </h5>
            </a>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-38 Header-actionsContainer-26"
        >
          <div
            class="makeStyles-container-39"
          >
            <button
              aria-controls="post-creation-menu"
              aria-haspopup="true"
              class="MuiButtonBase-root MuiButton-root makeStyles-button-40 MuiButton-contained"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label makeStyles-label-41"
              >
                <h5
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
                >
                  Створити...
                </h5>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-42"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header component tests header should be rendered without issues 1`] = `
<DocumentFragment>
  <div
    class="Header-header-1"
    id="header"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiContainer-root Header-container-3 MuiContainer-maxWidthLg"
    >
      <div
        class="Header-toolbarDesktop-5"
      >
        <div
          style="display: flex;"
        >
          <div
            class="MuiBox-root MuiBox-root-17"
          >
            <a
              href="/"
            >
              <h1
                class="MuiTypography-root Header-logo-6 MuiTypography-h1"
              >
                Доказові
              </h1>
            </a>
          </div>
          <div
            class="MuiBox-root MuiBox-root-18 Header-tabs-8"
          >
            <a
              aria-current="page"
              class="Header-tab-9 active"
              href="/"
            >
              <h5
                class="MuiTypography-root Header-tabLabel-10 MuiTypography-h5"
              >
                Головна
              </h5>
            </a>
            <a
              class="Header-tab-9"
              href="/materials"
            >
              <h5
                class="MuiTypography-root Header-tabLabel-10 MuiTypography-h5"
              >
                Матеріали
              </h5>
            </a>
            <a
              class="Header-tab-9"
              href="/experts"
            >
              <h5
                class="MuiTypography-root Header-tabLabel-10 MuiTypography-h5"
              >
                Автори
              </h5>
            </a>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-19 Header-actionsContainer-7"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Header component tests mobile version should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="Header-headerMobile-129"
    id="header"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiContainer-root Header-container-130 MuiContainer-maxWidthLg"
    >
      <div>
        <button
          aria-label="Open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-144 MuiIconButton-edgeStart"
          data-testid="menuButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-menuIcon-145"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root MuiBox-root-152"
      >
        <a
          href="/"
        >
          <h1
            class="MuiTypography-root Header-logoMobile-138 MuiTypography-h1"
          >
            Доказові
          </h1>
        </a>
      </div>
      <button
        aria-controls="simple-menu"
        aria-haspopup="true"
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root Header-searchIcon-139"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
